synchronization/supervized/supervisor.go

```diff
@@ -0,0 +1,72 @@
+package supervized
+
+import (
+	"context"
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/pkg/errors"
+	"runtime"
+	"strings"
+)
+
+type Errorer interface {
+	Error(message string, fields ...*log.Field)
+}
+
+func OneOff(logger Errorer, f func()) {
+	go func() {
+		defer recoverPanics(logger)
+		f()
+	}()
+}
+
+func LongLiving(ctx context.Context, logger Errorer, f func()) {
+	defer recoverPanics(logger)
+	go func() {
+		for {
+			f()
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// repeat
+				//TODO count restarts, fail if too many restarts, etc
+			}
+		}
+	}()
+}
+
+func recoverPanics(logger Errorer) {
```

Comment:
`recoverPanics` will effectively kill all stack traces on panics in the IDE when running tests.. It's an important mechanism for debugging..

Some alternatives to mitigate this:

1. Disable `recoverPanics` in test mode and only do this in production

2. When recovering from a panic, print a stack trace manually to log. Also, in testing mode, make sure the IDE identifies this stack trace and handles it like the default stack traces

Reply:
This is the main issue I want to think about together, all the others are cosmetic

Reply:
on the other hand, it allows us to understand where the panic comes from, no? that's the whole point of this function.

I'll add the stack trace to the log

Reply:
added